### PR TITLE
make `With` fields surrounding with `[``]`

### LINKED ISSUE #3: new sub logger using `with` can not get right log format
### what we do

```
func (s *testLogSuite) TestContextLogger(c *C)  {
	l, _, err := zaplog.InitLogger(&zaplog.Config{DisableStacktrace:true})
	if err != nil {
		panic(err)
	}
	l2 := l.With(zap.String("c", "c1"))
	l2.Error("", zap.String("k", "v"), zap.String("k2", "v2"))
}
```

### expect

```
[2019/03/04 23:00:05.943 +08:00] [ERROR] [log_test.go:209] [c=c1] [k=v] [k2=v2]
```

### but got 

```
[2019/03/04 23:00:05.943 +08:00] [ERROR] [log_test.go:209],c=c1 [k=v] [k2=v2]
```

### PR DESCRIPTION
**Pull request by lysu**
fixes #3 

because https://github.com/uber-go/zap/pull/685  can not be accepted in 1.x and  too many private method, so we temp copy whole ioCore to pingcap/log to call our `addFileds()` 

then we can make https://github.com/pingcap/tidb/pull/9548 works

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/log/4)

### CODE REVIEW

File: zap_text_core.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import "go.uber.org/zap/zapcore"
+
+// NewTextCore creates a Core that writes logs to a WriteSyncer.
+func NewTextCore(enc *textEncoder, ws zapcore.WriteSyncer, enab zapcore.LevelEnabler) zapcore.Core {
```
Comment (zz-jason) [inline zap_text_core.go]: how about:
```suggestion
func NewTextCore(enc *textEncoder, ws zapcore.WriteSyncer, enabler zapcore.LevelEnabler) zapcore.Core {
```

File: zap_text_core.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// it can be removed after https://github.com/uber-go/zap/pull/685 be merged
+type textIOCore struct {
+	zapcore.LevelEnabler
+	enc *textEncoder
+	out zapcore.WriteSyncer
+}
+
+func (c *textIOCore) With(fields []zapcore.Field) zapcore.Core {
+	clone := c.clone()
+	// it's different to ioCore, here call textEncoder#addFields to fix https://github.com/pingcap/log/pull/4
```
Comment (zz-jason) [inline zap_text_core.go]: Incorrect comment? This PR is https://github.com/pingcap/log/pull/4, and this comment claims that a function introduced in this PR is aimed to fix https://github.com/pingcap/log/pull/4 😂

File: zap_log_test.go [hunk 3]
```diff
@@ -153,15 +162,16 @@ func (t *testZapLogSuite) TestLog(c *C) {
...
-		`[INFO] [zap_log_test.go:118] ["Testing typs"] [filed1=noquote] `+
+		`[INFO] [zap_log_test.go:113] ["failed to fetch URL"] [url=http://example.com] [attempt=3] [backoff=1s]`,
+		`[INFO] [zap_log_test.go:118] ["failed to \"fetch\" [URL]: http://example.com"]`,
+		`[DEBUG] [zap_log_test.go:119] ["Slow query"] [sql="SELECT * FROM TABLE\n\tWHERE ID=\"abc\""] [duration=1.3s] ["process keys"=1500]`,
+		`[INFO] [zap_log_test.go:125] [Welcome]`,
+		`[INFO] [zap_log_test.go:126] ["Welcome TiDB"]`,
+		`[INFO] [zap_log_test.go:127] [欢迎]`,
+		`[INFO] [zap_log_test.go:128] ["欢迎来到 TiDB"]`,
+		`[WARN] [zap_log_test.go:129] [Type] [Counter=NaN] [Score=+Inf]`,
+		`[INFO] [zap_log_test.go:133] ["new connection"],[connID=1] [traceID=dse1121]`,
```
Comment (nolouch) [inline zap_log_test.go]: Why is there has a comma?

File: zap_text_core.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		return err
+	}
+	_, err = c.out.Write(buf.Bytes())
+	buf.Free()
+	if err != nil {
+		return err
+	}
+	if ent.Level > zapcore.ErrorLevel {
+		// Since we may be crashing the program, sync the output. Ignore Sync
+		// errors, pending a clean solution to issue #370.
```
Comment (zz-jason) [inline zap_text_core.go]: issue #370?

Comment (lysu) [reply zap_text_core.go]: - - this is old ioCore code..

File: zap_text_core.go (hunk 2 above)
Comment (lysu) [reply zap_text_core.go]: :rofl:

File: zap_text_core.go (hunk 1 above)
Comment (lysu) [reply zap_text_core.go]: maybe better keep old code name

File: zap_log_test.go (hunk 3 above)
Comment (lysu) [reply zap_log_test.go]: it seems we should change this https://github.com/pingcap/log/pull/4/files#diff-22234ef75ee1f622759c3325ab0a28deR434

File: zap_text_core.go (hunk 4 above)
Comment (nolouch) [reply zap_text_core.go]: How about add a link?

**Pull request merged**